tools/introspection/source/introspection_app.cpp

```diff
@@ -791,7 +792,10 @@ void IntrospectionApp::runIntrospection(const iox::units::Duration updatePeriod,
         }
     }
 
-    getchar();
+    iox::posix::posixCall(getchar)().failureReturnValue(EOF).evaluate().or_else([](auto& r) {
```

Comment:
@elfenpiff I had an error for dropping the return value, so I used it to report the error using `cout` I know you mentioned before that this might not be needed as `posixCall` take care of formatting a nice error message to the use, but I was using `iox::log::Logger`.

Is there a better way to do it ? handle the return value I mean?

Reply:
@zmostafa Yes there is! The method `expect` is exactly what you want here. If the call fails it prints the error message and terminates. So in this case:
```cxx
iox::posix::posixCall(getchar)().failureReturnValue(EOF).evaluate().expect("This error message should provide some context where the error is occurring");
```

Please try not to write error messages like this:
```cxx
std::cout << "getchar() call failed with EOF error" << r.getHumanReadableErrnum() << std::endl;
```

`posixCall` will print exactly this but in a more detailed fashion. It will provide file, line, the posix function which one has called, the return value, the errno value as integer and the errno as human readable string. So in the end when this would fail you would get the same error message more or less twice.

In every level you should provide some context where the error is occurring and state what is affected. I think in this case `getchar` is used to wait for user input to refresh something?! So it would be nice to state this. This would result in a nice and readable log output which reads somehow like a stack trace for humans.

In this case for instance:
```
introspection_app.cpp:795 (getchar) failed with errno XX, human readable errno message
unable to refresh whatever in introspection since getchar failed
```

The first line would come from the posix call the second would originate here. Now a fellow developer knows that getchar failed and that this lead to a failure in the introspection refresh thingy.

Reply:
@elfenpiff Thanks, I am using this [API page](https://iceoryx.io/v2.0.3/API-reference/hoofs/Files/posix__call_8hpp/#define-posixcall) to find about `posixCall` and search in the code to find usage examples,and could not find anything about using `expect`.

I think we need to update the page with what you explained me already, this is more than what is written there. :sweat_smile:

Reply:
@zmostafa You are right. In the end the `posixCall` returns an `expected` and this provides the method `expect`